server/src/main/java/io/druid/guice/DruidProcessingModule.java

```diff
@@ -98,6 +98,7 @@ public ExecutorService getProcessingExecutorService(
             lifecycle,
             config
         ),
+        getMergeBufferPool(config),
```

Comment:
This shouldn't be `getMergeBufferPool(config)` - it's going to create a new merge buffer pool rather than use the existing one. (Check the implementation of getMergeBufferPool.) So the metric would always be zero since it wouldn't be checking the "real" pool. Instead, the patch should have a `@Merging BlockingPool<ByteBuffer> mergePool` as an argument to `getProcessingExecutorService`. Then the right one would get injected.

However! I think we should go a different path entirely. The Processing ExecutorService is not really the right place to add this metric. The metric is unrelated to executor processing. I think it'd be better to have the Merging BlockingPool do this itself, in much the same way that the Processing ExecutorService does. By that I mean: define a MetricsEmittingBlockingPool and have that emit the buffersUsed metric. Then, that should be returned by the existing `getMergeBufferPool` method.

Reply:
Hey @gianm.

Thanks for your comments. I also had doubt while using this function but then I guessed that @LazySingleton annotation will take care of that. 

No worries. I will fix this thing and let you know in sometime.